sound/soc/sof/ipc4-topology.c

```diff
@@ -1145,6 +1179,12 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 	case snd_soc_dapm_dai_in:
 	case snd_soc_dapm_dai_out:
 	{
+		struct snd_sof_widget *pipe_widget = swidget->pipe_widget;
+		struct sof_ipc4_pipeline *pipeline = pipe_widget->private;
+
+		if (pipeline->use_chain_dma)
```

Comment:
I think we should add a check here to make sure that only HDA DAI's are using the chained DMA. In fact prepare() might be a bit too late, we should do this in sof_ipc4_widget_setup_comp_dai(). The tricky part will be to make sure that the pipeline widgets are parsed first. I have a feeling they are but best to check.

Reply:
BTW are you missing something in the unprepare_copier_module()

Reply:
I dropped them on purpose. I found it clearer and more fool proof (me being the fool not being certain in which order the functions are called), to zero the dma id bit right before assigning new values there, like I do with scs bit and the fifo_size. Felt unsymmetrical doing the zeroing differently with different bits. But if you think its better to go back to old and maybe even drop the zeroing before the assignment in dma id cases, then sure I'll go back.

Reply:
I think it would be more balanced to set the bits in prepare() and clear them in unprepare()

Reply:
> I think we should add a check here to make sure that only HDA DAI's are using the chained DMA. In fact prepare() might be a bit too late, we should do this in sof_ipc4_widget_setup_comp_dai(). The tricky part will be to make sure that the pipeline widgets are parsed first. I have a feeling they are but best to check.

@ranj063 I've addressed all the other comments, but this is a bit tricky as in sof_ipc4_widget_setup_comp_dai()  the associated pipeline object is not yet built, so I do not know any easy way to dig out the use_chain_dma property. So I assume I should do the check in sof_ipc4_prepare_copier_module() after all, or is there something I overlooked?